### ISSUE TITLE
Slow speeds and connection errors

### ISSUE THREAD

**Issue by fkoemep** [Jigsaw-Code/Intra#508]
Whenever I try to set the custom server https://dns11.quad9.net the connection simply stops working. 

Also no matter which dns provider I choose, I get extremely slow speeds. My connection is 300mbps and I'm barely getting 50mbps download, as soon as I disable Intra the speeds go back to normal, this happens both on wifi and data.

**Comment by devashishkatoriya** [comment]
I'm also observing same issue. My broadband is 100 mbps. However the moment I turn on Intra (any dns server), the speed falls to about 45-50 mbps.

The speed again restores if I disable Intra.

**Comment by ignoramous** [comment]
These may fix the issues you've reported (it fixes them on a hard fork of Intra I maintain). Perhaps, these fixes land in the next Intra release.
- #519 
- #517

**Comment by fkoemep** [comment]
Any news when the next version is going to be released? It's been 4 months already

**Comment by berksmbl** [comment]
Hi @jyyi1 #526 also seems to have an update regarding this issue. When will this be released?

**Comment by fkoemep** [comment]
I updated to Intra v1.4.1 today but I still face the same issues, with my normal connection I achieve 300mbps+ of download speed compared to an average of 50mbps when I'm using Intra 😕

**Comment by jyyi1** [comment]
Thanks for the report!

We've confirmed the decrease in connection speed. Our team is currently heads-down on another Intra-related project (which will support a lot more platforms than just Android, news coming soon!).

We'd really appreciate any help from the community in investigating and addressing this speed issue in the repository. Contributions are welcome!

**Comment by fkoemep** [comment]
> Thanks for the report!
> 
> We've confirmed the decrease in connection speed. Our team is currently heads-down on another Intra-related project (which will support a lot more platforms than just Android, news coming soon!).
> 
> We'd really appreciate any help from the community in investigating and addressing this speed issue in the repository. Contributions are welcome!

Does this project maybe have something to do with this issue? https://issuetracker.google.com/issues/331250145?pli=1

According to Google it's blocked by another issue which we don't have access to. If you could do anything, no matter how little, to make this happen it would be a game changer for Android!

**Comment by jyyi1** [comment]
> > Thanks for the report!
> > We've confirmed the decrease in connection speed. Our team is currently heads-down on another Intra-related project (which will support a lot more platforms than just Android, news coming soon!).
> > We'd really appreciate any help from the community in investigating and addressing this speed issue in the repository. Contributions are welcome!
> 
> Does this project maybe have something to do with this issue? https://issuetracker.google.com/issues/331250145?pli=1
> 
> According to Google it's blocked by another issue which we don't have access to. If you could do anything, no matter how little, to make this happen it would be a game changer for Android!

The new project isn't directly tied to the issue you linked, which appears to be an Android-only implementation. Our focus is broader. We are aiming to support for all major platforms, including Android, iOS, Windows, and others.

**Comment by Lanius-collaris** [comment]
@jyyi1 Can you try tuning some options of lwip? I saw heiher [did](https://github.com/heiher/lwip/blob/main/src/ports/include/lwipopts.h) this and got good result. Here is some macros with very small default value:
```
PBUF_POOL_SIZE
TCP_SND_BUF
TCP_WND
MEMP_NUM_UDP_PCB
MEMP_NUM_PBUF
MEMP_NUM_TCP_PCB
```
